pkg/didcomm/protocol/messagepickup/service_test.go

```diff
@@ -0,0 +1,906 @@
+	t.Run("test inbox.DecodeMessages() - error", func(t *testing.T) {
+		ibx := &inbox{
+			Messages: nil,
+		}
+
+		_, err := ibx.DecodeMessages()
+		require.NoError(t, err)
+	})
+}
+
+func TestHandleOutbound(t *testing.T) {
+	t.Run("test MessagePickupService.HandleOutbound() - not implemented", func(t *testing.T) {
+		svc, err := getService()
+		require.NoError(t, err)
+
+		svc.connectionLookup = &connectionsStub{
+			getConnRecord: func(string) (*connection.Record, error) {
+				return nil, storage.ErrDataNotFound
+			},
+		}
+
+		err = svc.HandleOutbound(nil, "not", "implemented")
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "not implemented")
+	})
+}
+func TestPullMessages(t *testing.T) {
+	t.Run("test MessagePickupService.pullMessages() - get inbox error", func(t *testing.T) {
+		mockStore := mockstore.NewMockStoreProvider()
+		svc, err := New(&mockprovider.Provider{
+			StorageProviderValue:          mockStore,
+			TransientStorageProviderValue: mockstore.NewMockStoreProvider(),
+			OutboundDispatcherValue:       nil,
+		}, &mockTransportProvider{
+			packagerValue: &mockPackager{},
+		})
+		require.NoError(t, err)
+
+		mockStore.Store.ErrGet = errors.New("error get")
+
+		_, err = svc.pullMessages(THEIRDID, 1)
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "error get")
+	})
+
+	t.Run("test MessagePickupService.pullMessages() - put inbox error", func(t *testing.T) {
+		mockStore := mockstore.NewMockStoreProvider()
+		svc, err := New(&mockprovider.Provider{
+			StorageProviderValue:          mockStore,
+			TransientStorageProviderValue: mockstore.NewMockStoreProvider(),
+			OutboundDispatcherValue:       nil,
+		}, &mockTransportProvider{
+			packagerValue: &mockPackager{},
+		})
+		require.NoError(t, err)
+
+		b, err := json.Marshal(&inbox{DID: THEIRDID})
+		require.NoError(t, err)
+
+		err = mockStore.Store.Put(THEIRDID, b)
+		require.NoError(t, err)
+
+		mockStore.Store.ErrPut = errors.New("error put")
+
+		_, err = svc.pullMessages(THEIRDID, 1)
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "error put")
+	})
+}
+
+func TestGetConnection(t *testing.T) {
+	t.Run("test MessagePickupService.getConnection() - error", func(t *testing.T) {
+		svc, err := getService()
+		require.NoError(t, err)
+
+		svc.connectionLookup = &connectionsStub{
+			getConnRecord: func(string) (*connection.Record, error) {
+				return nil, storage.ErrDataNotFound
+			},
+		}
+
+		_, err = svc.getConnection("test")
+		require.Error(t, err)
+		require.True(t, errors.Is(err, ErrConnectionNotFound))
+	})
+}
+
+func getService() (*Service, error) {
+	svc, err := New(&mockprovider.Provider{
+		StorageProviderValue:          mockstore.NewMockStoreProvider(),
+		TransientStorageProviderValue: mockstore.NewMockStoreProvider(),
+		OutboundDispatcherValue:       nil,
+	}, &mockTransportProvider{
+		packagerValue: &mockPackager{},
+	})
+
+	return svc, err
+}
+
+// mockProvider mock provider
+type mockTransportProvider struct {
```

Comment:
Why not to use `gomock` ?

Reply:
@soluchok this mocking is fine. I'm not a fan of gomock. It's slowing down the builds and there is *baaaarely* any return on the investment.